import_flow_events.py

```diff
@@ -89,23 +89,31 @@
       -- but redshift doesn't support that.
       flow_time BIGINT NOT NULL ENCODE lzo,
       flow_id VARCHAR(64) NOT NULL DISTKEY ENCODE lzo,
-      type VARCHAR(64) NOT NULL ENCODE lzo
+      type VARCHAR(64) NOT NULL ENCODE lzo,
+      export_date DATE NOT NULL ENCODE lzo
     );
 """
 
 Q_CHECK_FOR_DAY = """
     SELECT timestamp FROM flow_events
-    WHERE timestamp::DATE = '{day}'::DATE
+    WHERE timestamp::DATE <= '{day}'::DATE + '1 day'::INTERVAL
+      AND timestamp::DATE >= '{day}'::DATE - '1 day'::INTERVAL
+      AND export_date = '{day}'::DATE;
     LIMIT 1;
 """
 
-Q_CLEAR_DAY_METADATA = """
-    DELETE FROM flow_metadata
-    WHERE begin_time::DATE = '{day}'::DATE;
-"""
 Q_CLEAR_DAY_EVENTS = """
     DELETE FROM flow_events
-    WHERE timestamp::DATE = '{day}'::DATE;
+    WHERE timestamp::DATE <= '{day}'::DATE + '1 day'::INTERVAL
+      AND timestamp::DATE >= '{day}'::DATE - '1 day'::INTERVAL
+      AND export_date = '{day}'::DATE;
+"""
+Q_CLEAR_DAY_METADATA = """
+    DELETE flow_metadata
+    FROM flow_metadata AS m
+    LEFT JOIN flow_events AS e
+      ON m.flow_id = e.flow_id
+    WHERE e.flow_id IS NULL;
```

Comment:
> Would that lead to (a small number of) duplicated events at the boundaries?

Maybe.  If you've got a concrete example scenario in mind it would be great to write it out explicitly.

I see two possibilities - the events from the start of a flow being deleted and re-imported, and events from the end of a flow being deleted and re-imported.

If we're deleting events from the end of the flow, then we are not deleting the `flow.begin` event and so the flow will just stick around in the `flow_metadata` table.  We might re-calculate its various properties like "completed" and "new_account", but we won't add a duplicate entry for it.

If we're deleting events from the start of the flow, then we delete the `flow.begin` event and hence delete the row from `flow_metadata`.  It will be re-created when we re-import the events for that day.  But we have to be careful to ensure that it gets the appropriate values for "completed", "new_account" etc, because those events might have come from the following day, and are not being re-imported.